### PR TITLE
[nrf noup] include: net: add secure socket options

### PR DESCRIPTION
**Pull request by lemrey**
Add secure socket options:
* Add CID socket options to NCS specific options.
* Add TLS/DTLS tls ciphersuite chosen socket option to NCS specific options.
* Add TLS/DTLS connection save/load socket options to NCS specific options.
* Add TLS/DTLS handshake status socket option to NCS specific options.

### CODE REVIEW

File: include/zephyr/net/socket_ncs.h [hunk 1]
```diff
@@ -34,11 +34,39 @@ extern "C" {
...
 
+/** Socket option to set and get DTLS CID setting, specifically for nRF sockets.
+ *  The option accepts an integer, indicating the setting.
+ *  Accepted vaules for the option are: 0, 1 and 2.
+ */
+#define TLS_DTLS_CID 17
+
+/** Socket option to get DTLS CID status, specifically for nRF sockets.
+ *  The option accepts an integer, indicating the setting.
+ *  Accepted vaules for the option are: 0, 1 and 2.
```
Comment (eivindj-nordic) [inline include/zephyr/net/socket_ncs.h:45]:
```suggestion
 *  Accepted vaules for the option are: 0, 1, 2 and 3.
```

**Comment by rlubos** [comment]
Can you add `squash! [nrf noup] include: net: add NCS extensions` line on top of the commit body?

File: include/zephyr/net/socket_ncs.h [hunk 2]
```diff
@@ -34,11 +34,39 @@ extern "C" {
...
+#define TLS_DTLS_CONN_SAVE 19
+
+/** Socket option to load DTLS connection, specifically for nRF sockets.
+ */
+#define TLS_DTLS_CONN_LOAD 20
+
+/** Socket option to get result of latest TLS/DTLS completed handshakes end status,
+ *  specifically for nRF sockets.
+ *  The option accepts an integer, indicating the setting.
+ *  Accepted vaules for the option are: 0, 1 and 2.
```
Comment (eivindj-nordic) [inline include/zephyr/net/socket_ncs.h:60]:
```suggestion
 *  Accepted vaules for the option are: 0 and 1.
```

File: include/zephyr/net/socket_ncs.h (hunk 1 above)
Comment (eivindj-nordic) [reply include/zephyr/net/socket_ncs.h:45]: We might also just remove this sentence, as the options are given below. Same with the CID and handshake status.

**Review by eivindj-nordic — APPROVED** [review]

**Pull request merged**